### PR TITLE
Update doc in GitOps related sections for 0.6.0 release

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/578#" title="Author self-approved">jiayiwang7</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[docs/OWNERS](https://github.com/aws/eks-anywhere/blob/main/docs/OWNERS)~~ [jiayiwang7]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: docs/content/en/docs/tasks/cluster/cluster-upgrades.md
Comment (danbudris) [inline docs/content/en/docs/tasks/cluster/cluster-upgrades.md:47]: "Most upgrade changes are backwards compatible." 
as a user, this raises more questions for me than it answers. Do we have a list of which are backward compatible and which are not? If this isn't something we clearly supply maybe remove this phrase

File: docs/content/en/docs/tasks/cluster/cluster-upgrades.md
Comment (danbudris) [inline docs/content/en/docs/tasks/cluster/cluster-upgrades.md:51]:
```suggestion
In order to accommodate the management cluster feature, the CLI will now structure the repo directory following a new convention:
```

File: docs/content/en/docs/tasks/cluster/cluster-upgrades.md
Comment (danbudris) [inline docs/content/en/docs/tasks/cluster/cluster-upgrades.md:68]:
```suggestion
*By default, Flux kustomization reconciles at the management cluster's root level (`./clusters/management-cluster`), so both the management cluster and all of the workload clusters it manages are synced.*
```

File: docs/content/en/docs/tasks/cluster/cluster-upgrades.md
Comment (danbudris) [inline docs/content/en/docs/tasks/cluster/cluster-upgrades.md:70]: I think the first information we present after ["GitOps Controller Upgrade"](https://github.com/aws/eks-anywhere/pull/578/files#diff-34ef9f9e4b2404d53a4b13741331d54859a0195da2bdca57ae73e83d3626aa06R49) should be a bolded or note-boxed call-out which draws the users attention to the fact that we need to perform a manual action before upgrading; I think this should be before displaying the new convention and explaining how to change  the dir structure for existing repos

File: docs/content/en/docs/tasks/cluster/cluster-upgrades.md
Comment (danbudris) [inline docs/content/en/docs/tasks/cluster/cluster-upgrades.md:84]:
```suggestion
After pushing the changes to the Git repo, user can safely run the `upgrade` command, and the Flux components will be upgraded.
```

File: docs/content/en/docs/tasks/cluster/cluster-upgrades.md
Comment (danbudris) [reply docs/content/en/docs/tasks/cluster/cluster-upgrades.md:70]:
```suggestion
__Upgrading an existing GitOps enabled cluster created by an older release of the EKS-A CLI (v0.5.0 and below) requires a manual update to the git repository structure.__ 

Before running the `upgrade` command on an existing GitOps enabled cluster, you must:
- move the eksa cluster config into a subdirectory under the management cluster's root level (as shown above).
- update the `clusterConfigPath` field of the `GitOpsConfig` of the `eksa-cluster.yaml` to reflect the new path of the `eksa-system` directory. For example:
```

**Review by danbudris — CHANGES_REQUESTED** [review]
Two big things I think we should do here:
- really clearly call out the required manual upgarde in bold or note box at the top of the section starting at line 49 section

- use a list (even if it is two items) to clearly explain what needs to happen, and provide examples (which we do with the tree of the structure)

File: docs/content/en/docs/tasks/cluster/cluster-upgrades.md
Comment (jiayiwang7) [reply docs/content/en/docs/tasks/cluster/cluster-upgrades.md:47]: how about "All upgrade changes are backwards compatible expect GitOps"?

File: docs/content/en/docs/tasks/cluster/cluster-upgrades.md
Comment (jiayiwang7) [reply docs/content/en/docs/tasks/cluster/cluster-upgrades.md:70]: jesus only by comparison do i realize how bad my writing is. yours so short but clear

File: docs/content/en/docs/tasks/cluster/cluster-upgrades.md
Comment (danbudris) [reply docs/content/en/docs/tasks/cluster/cluster-upgrades.md:47]: yeah that makes sense

**Review by danbudris — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/578#pullrequestreview-798254968" title="Approved">danbudris</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/578#" title="Author self-approved">jiayiwang7</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[docs/OWNERS](https://github.com/aws/eks-anywhere/blob/main/docs/OWNERS)~~ [danbudris,jiayiwang7]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
